pkg/gardenlet/controller/shoot/shoot_control.go

```diff
@@ -52,12 +52,16 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/reconcile"
 )
 
-func (c *Controller) shootAdd(obj interface{}) {
+func (c *Controller) shootAdd(obj interface{}, resetRateLimiting bool) {
 	key, err := cache.MetaNamespaceKeyFunc(obj)
 	if err != nil {
 		logger.Logger.Errorf("Couldn't get key for object %+v: %v", obj, err)
 		return
 	}
+
+	if resetRateLimiting {
+		c.getShootQueue(obj).Forget(key)
```

Comment:
Is resetting really necessary? IIUC, `.Add(key)` adds the element immediately to the queue and `.AddRateLimited(key)` with a delay.

Reply:
Hm, I haven't studied all the internals of these queues, but if this would be true then this PR would be unnecessary.
The doc string of `Forget` states:

```golang
// Forget indicates that an item is finished being retried.  Doesn't matter whether it's for perm failing
// or for success, we'll stop the rate limiter from tracking it.  This only clears the `rateLimiter`, you
// still have to call `Done` on the queue.
```

I'm not calling `Done()` because a worker might be in the process of reconciling the shoot (my understanding is that if I'd be calling `Done()` then it might happen that two workers can simultaneously process a shoot). `Forget()` is doing exactly what I want - clearing the rate limiter.